crates/red_knot_python_semantic/resources/mdtest/strings.md

```diff
@@ -0,0 +1,129 @@
+# Strings
+
+## Literals
+
+### Simple
+
+We can infer a string literal type and track concatenated string literals as well:
+
+```py
+w = "Hello"
+x = 'world'
+y = "Guten " + 'tag'
+z = 'bon ' + "jour"
+
+reveal_type(w)  # revealed: Literal["Hello"]
+reveal_type(x)  # revealed: Literal["world"]
+reveal_type(y)  # revealed: Literal["Guten tag"]
+reveal_type(z)  # revealed: Literal["bon jour"]
+```
+
+### Nested Quotes
+
+We can handle string literals with nested quotes:
+
+```py
+x = 'I say "hello" to you'
+y = "You say \"hey\" back"
+z = 'No "closure here'
+reveal_type(x)  # revealed: Literal["I say \"hello\" to you"]
+reveal_type(y)  # revealed: Literal["You say \"hey\" back"]
+reveal_type(z)  # revealed: Literal["No \"closure here"]
+```
+
+## f-strings
```

Comment:
I'd rather put f-strings in their own file; it can either be a top-level `fstrings.md` or we can add a `literal/` subdir and put several of these files in there.